libs/csrf/csrfprotector.php

```diff
@@ -106,52 +106,52 @@ public static function init($length = null, $action = null)
 
 			/*
 			 * load configuration file and properties
-			 * Check locally for a config.php then check for 
+			 * Check locally for a config.php then check for
 			 * a config/csrf_config.php file in the root folder
 			 * for composer installations
 			 */
 			$standard_config_location = __DIR__ ."/../config.php";
 			$composer_config_location = __DIR__ ."/../../../../../config/csrf_config.php";
 
 			if (file_exists($standard_config_location)) {
-				self::$config = include($standard_config_location);
+				static::$config = include($standard_config_location);
```

Comment:
You're right. For my purposes, I think only `logCSRFattack` and `failedValidationAction` need to be extended. Let me test to make sure and then I can push the more limited update for your review.